aiokafka/conn.py

```diff
@@ -59,23 +105,40 @@ def _read(self):
 
                 resp = yield from self._reader.readexactly(size)
 
-                fut = self._requests.pop(0)
-                if not fut.cancelled():
-                    fut.set_result(resp)
+                recv_correlation_id, = self.HEADER.unpack(resp[:4])
+
+                correlation_id, resp_type, fut = self._requests.pop(0)
+                if (self._api_version == (0, 8, 2)
+                        and resp_type is GroupCoordinatorResponse
+                        and correlation_id != 0 and recv_correlation_id == 0):
+                    self.log.warning(
+                        'Kafka 0.8.2 quirk -- GroupCoordinatorResponse'
+                        ' coorelation id does not match request. This'
+                        ' should go away once at least one topic has been'
+                        ' initialized on the broker')
+
+                if correlation_id != recv_correlation_id:
+                    error = Errors.CorrelationIdError(
+                        'Correlation ids do not match: sent {}, recv {}'
+                        .format(correlation_id, recv_correlation_id))
+                    fut.set_exception(error)
```

Comment:
This is kinda strange. Is it possible, that after the correlation does not match once, we can continue to work with this connection?